apps/login/src/app/(login)/idp/[provider]/success/page.tsx

```diff
@@ -11,191 +11,158 @@ import {
-                return (
-                  <DynamicTheme branding={branding}>
-                    <div className="flex flex-col items-center space-y-4">
-                      <h1>Linking failed</h1>
-                      <div className="w-full">
-                        {
-                          <Alert type={AlertType.ALERT}>
-                            {JSON.stringify(error.message)}
-                          </Alert>
-                        }
-                      </div>
-                    </div>
-                  </DynamicTheme>
-                );
-              });
-
-              if (idpLink) {
-                return (
-                  // TODO: possibily login user now
-                  <DynamicTheme branding={branding}>
-                    <div className="flex flex-col items-center space-y-4">
-                      <h1>Account successfully linked</h1>
-                      <div>Your account has successfully been linked!</div>
-                    </div>
-                  </DynamicTheme>
-                );
-              }
-            }
-          }
-
-          if (options?.isCreationAllowed && options.isAutoCreation) {
-            const newUser = await createUser(providerType, idpInformation);
-
-            if (newUser) {
-              return (
-                <DynamicTheme branding={branding}>
-                  <div className="flex flex-col items-center space-y-4">
-                    <h1>Register successful</h1>
-                    <div>You have successfully been registered!</div>
-                  </div>
-                </DynamicTheme>
-              );
-            }
-          }
-
-          // return login failed if no linking or creation is allowed and no user was found
-          return (
-            <DynamicTheme branding={branding}>
-              <div className="flex flex-col items-center space-y-4">
-                <h1>Login failed</h1>
-                <div className="w-full">
-                  {
-                    <Alert type={AlertType.ALERT}>
-                      User could not be logged in
-                    </Alert>
-                  }
-                </div>
-              </div>
-            </DynamicTheme>
-          );
-        } else {
-          return (
-            <DynamicTheme branding={branding}>
-              <div className="flex flex-col items-center space-y-4">
-                <h1>Login failed</h1>
-                <div className="w-full">
-                  {
-                    <Alert type={AlertType.ALERT}>
-                      Could not get user information
-                    </Alert>
-                  }
-                </div>
-              </div>
-            </DynamicTheme>
-          );
-        }
-      })
-      .catch((error) => {
+
+  if (!provider || !id || !token) {
+    return loginFailed(branding);
+  }
+
+  const intent = await retrieveIDPIntent(id, token);
+
+  const { idpInformation, userId } = intent;
+
+  if (userId && id && token) {
```

Comment:
id and toke are never falsy, see https://github.com/zitadel/typescript/pull/217/files#r1797378651

Reply:
yeah, but these are not from the api, these are the search params from the url which can be undefined so we have to add this check

Reply:
?

```suggestion
  if (userId) {
```

Reply:
ah now I get it.. I've removed the redundant check.. thx 👍